client/components/masonry-wave/index.jsx

```diff
@@ -0,0 +1,91 @@
+import './style.scss';
+
+import { useState, useEffect } from 'react';
+
+function getWindowDimensions() {
+	const { innerWidth: width, innerHeight: height } = window;
+	return {
+		width,
+		height,
+	};
+}
+
+function useWindowDimensions() {
+	const [ windowDimensions, setWindowDimensions ] = useState( getWindowDimensions() );
+
+	useEffect( () => {
+		function handleResize() {
+			setWindowDimensions( getWindowDimensions() );
+		}
+
+		window.addEventListener( 'resize', handleResize );
```

Comment:
The viewport package has helpers to subscribe to changes in size, but we might want to add some throttling/debouncing to this.

Reply:
Good catch in here but will probably use https://github.com/Automattic/wp-calypso/tree/trunk/packages/viewport-react !